cli/tests/default/unit/test_engine_type.py

```diff
@@ -2,79 +2,127 @@
 
 from semgrep.app.scans import ScanHandler
 from semgrep.engine import EngineType as ET
+from semgrep.error import SemgrepError
 from semgrep.meta import GitMeta
 
 
 @pytest.mark.quick
+@pytest.mark.parametrize("is_supply_chain_only", [True, False])
+@pytest.mark.parametrize("is_secrets_scan", [True, False])
 @pytest.mark.parametrize(
-    ("is_cloud_flag_on", "is_ci_scan_full", "enable_pro_diff", "requested", "expected"),
+    "engine_flag", [None, ET.OSS, ET.PRO_LANG, ET.PRO_INTRAFILE, ET.PRO_INTERFILE]
+)
+@pytest.mark.parametrize(
+    (
+        "logged_in",
+        "is_interfile_flag_on",
+        "is_ci_scan_full",
+        "interfile_diff_scan_enabled",
+        "expected_default",
+    ),
     [
+        # These combinations define the expected engine for Code scans when no
+        # engine is explicitly requested via a CLI argument. Requesting the engine
+        # in the CLI or running Secrets or Supply Chain may change the engine used.
+        #
+        # We assume some invariants and thus don't test them:
+        # - not logged_in -> is_interfile_flag_on is None, is_ci_scan_full is None
+        # - is_interfile_flag_on is None -> is_ci_scan_full is None
+        #
         # semgrep scan
-        (False, None, False, None, ET.OSS),
-        (False, None, False, ET.OSS, ET.OSS),
-        (False, None, False, ET.PRO_LANG, ET.PRO_LANG),
-        (False, None, False, ET.PRO_INTRAFILE, ET.PRO_INTRAFILE),
-        (False, None, False, ET.PRO_INTERFILE, ET.PRO_INTERFILE),
+        (False, None, None, False, ET.OSS),
+        (False, None, None, True, ET.OSS),
+        (True, None, None, False, ET.OSS),
```

Comment:
Why do we expect `OSS` here even though we are logged in ?

Reply:
I guess it's because it's `semgrep scan`, although I'd say we should make `--pro-intrafile` the default also for `semgrep scan`.

Reply:
My only concern is the need to download the pro binary; seems like a bad first experience. If we speed up the download then I agree.

Reply:
Something to schedule for Q2 maybe. I suspect that PR will also come with a bunch of testing considerations. Will bring it up in the RICE session